### DISCUSSION TITLE
Advanced Search - Results are "AND" and not "OR".  Any way around this?

### DISCUSSION THREAD

**Discussion by jcurry5** [nextgenhealthcare/connect#5807] [Q&A] [answered]
I need to search for multiple records and reprocess them which is easier to do by searching them all at once and instead of one at a time.

I have metadata enabled for the values but the advanced search is only doing an AND and I'm not see any other way to change this to an OR.  Any other way around this or am I stuck searching one record at a time?

**Comment by jonbartels** [comment, marked as answer]
The only workaround I can think of is to:

1. Replicate the SQL query with the appropriate OR conditions in a DB reader, if you're querying on metadata you would likely need the `d_##mcm` table where `##` is the LOCAL channel ID
2. Use the message IDs from this DB reader to call the MC API for `/channels/{channelId}/messages/{messageId}/_reprocess`

You could use either internal APIs or the HTTP API

**Reply by jcurry5** [reply]
Thanks - interesting way of doing it.

Do you know if there is anything in the backlog to add additional search conditions?

**Reply by jonbartels** [reply]
Not offhand. I try to contribute ideas and PRs to the backlog but I can't keep up with the details of the backlog.

One other workaround - If you want to reprocess on an OR condition can you run mutliple searches constrained by message status or do you have to maintain order?

**Reply by jcurry5** [reply]
Message order isn't of importance as they will be unique records.  Appreciate the response.
